feat(hover-state-desc): add hover state descriptions to icon buttons

orbitalsqwib opened the pull request:
## Problem

Currently there is no description of each icon when users hovers over it. To find out what the button/icon does, the user has to click on it first. This is not a good user experience.

Closes #263

## Solution

- Added tooltips for share checker, copy share links actions
- Added tooltips for dashboard duplicate and delete actions
- Added tooltips for builder field duplicate and delete actions
- Added tooltips for builder constant map delete actions
- Added tooltips for logic show/hide actions
- Added tooltips for logic conditional add and delete actions
- Added 800ms of opening delay before tooltips are shown

## Screenshots
<details>
<summary>Click here to show sample screenshots</summary>
<img width="1792" alt="Screenshot 2021-04-13 at 1 37 36 PM" src="https://user-images.githubusercontent.com/21305518/114504919-2f58aa00-9c62-11eb-9c15-b4fa084d0eda.png">
<img width="1792" alt="Screenshot 2021-04-13 at 2 09 28 PM" src="https://user-images.githubusercontent.com/21305518/114504925-31bb0400-9c62-11eb-886d-a0a3cda4283f.png">
<img width="1792" alt="Screenshot 2021-04-13 at 2 09 39 PM" src="https://user-images.githubusercontent.com/21305518/114504932-341d5e00-9c62-11eb-9559-d3d2787dc679.png">
<img width="1792" alt="Screenshot 2021-04-13 at 2 09 47 PM" src="https://user-images.githubusercontent.com/21305518/114504935-354e8b00-9c62-11eb-8f56-5bacadce6e72.png">
<img width="1792" alt="Screenshot 2021-04-13 at 2 09 58 PM" src="https://user-images.githubusercontent.com/21305518/114504940-367fb800-9c62-11eb-824c-120580eac641.png">
<img width="1792" alt="Screenshot 2021-04-13 at 2 10 09 PM" src="https://user-images.githubusercontent.com/21305518/114504942-37184e80-9c62-11eb-87bd-89d97cbec7cc.png">
<img width="1790" alt="Screenshot 2021-04-13 at 2 10 20 PM" src="https://user-images.githubusercontent.com/21305518/114504946-37b0e500-9c62-11eb-8405-aa4ff428c6cf.png">
<img width="1792" alt="Screenshot 2021-04-13 at 2 10 30 PM" src="https://user-images.githubusercontent.com/21305518/114504947-38497b80-9c62-11eb-9cfd-c7e16768785e.png">
<img width="1792" alt="Screenshot 2021-04-13 at 2 10 45 PM" src="https://user-images.githubusercontent.com/21305518/114504949-38e21200-9c62-11eb-8c4b-42b550348a3b.png">
<img width="1792" alt="Screenshot 2021-04-13 at 2 10 53 PM" src="https://user-images.githubusercontent.com/21305518/114504953-397aa880-9c62-11eb-90fc-851d7cdc779d.png">


</details>

## Tests

- [ ] Check that tooltips display for icons on dashboard (delete, duplicate)
- [ ] Check that tooltips display for sharing modal (share, copy link)
- [ ] Check that tooltips display in builder fields (delete, duplicate)
- [ ] Check that tooltips display in builder constant mapping modal (delete mapping)
- [ ] Check that tooltips display in builder logic fields (show result, hide result, delete, duplicate)
- [ ] Check that tooltips display in builder conditional logic fields (add, delete conditional)
- [ ] Check that all tooltips have a delay before showing up on hover

prestonlimlianjie commented on src/client/components/builder/EmbedModal.tsx:
```diff
@@ -49,7 +51,17 @@ const EmbedField: FC<EmbedFieldProps> = ({ name, value, children }) => {
+            >
+              <BiCopy />
+            </ReactIconTooltip>
+          }
```
```suggestion
          children={
            <Tooltip
              label="Copy"
              placement="right"
              openDelay={800}
              gutter={16}
              hasArrow
            >
              <span><BiCopy /></span>
            </Tooltip>
          }
```

Would this work? If so, we can do without the custom `ReactIconTooltip` component and only use the `Tooltip` Chakra component.

prestonlimlianjie reviewed (commented):
Thanks for the quick PR! The PR description is very detailed and helpful. 

I have a request - could we remove `ReactIconTooltip` if we can do without it? It would be nice to use 1 version of tooltip throughout the codebase (i.e. `Tooltip` from chakra). I've made a suggested change.

Could I also trouble you to run @NatMaeTan through the various labels? I suspect she might want to standardize whether we are using verbs (e.g. `Delete` or `Copy`) or adjectives (e.g. `Shown` or `Hidden`) for our tooltip labels.

Also in the Constants tab, under the map table, we have two instances of `Delete`; one to signify `Delete row` and one to signify `Delete table`. I'm wondering if Nat has suggestions on differentiating these two labels.

Commit:
feat(hover-desc): add hover state descriptions to icon buttons

- Add wrapper component for chakra tooltips implementing
default props
- Add tooltips for share checker, copy share links actions
- Add tooltips for dashboard duplicate and delete actions
- Add tooltips for checker field duplicate and delete actions
- Add tooltips for question option delete actions
- Add tooltips for checker constant map delete actions
- Add tooltips for logic show/hide actions
- Add tooltips for logic conditional add and delete actions

Commit:
feat(hover-desc): fix copy for tooltips

prestonlimlianjie reviewed (approved):
lgtm!
